x-pack/plugins/maps/server/maps_telemetry/telemetry_task.js

```diff
@@ -13,12 +13,16 @@ export const TASK_ID = `Maps-${TELEMETRY_TASK_TYPE}`;
 export function scheduleTask(server, taskManager) {
   const { kbnServer } = server.plugins.xpack_main.status.plugin;
 
-  kbnServer.afterPluginsInit(() => {
-    taskManager.schedule({
-      id: TASK_ID,
-      taskType: TELEMETRY_TASK_TYPE,
-      state: { stats: {}, runs: 0 },
-    });
+  kbnServer.afterPluginsInit(async () => {
+    try {
+      await taskManager.schedule({
+        id: TASK_ID,
+        taskType: TELEMETRY_TASK_TYPE,
+        state: { stats: {}, runs: 0 },
+      });
+    }catch(e) {
+      server.log(['warning', 'maps'], `Error scheduling telemetry task, received ${e.message}`);
```

Comment:
Couple things:

1. This is using the tag `maps`, and I assume it should be using a tag relevant to task manager.
2. I don't think swallowing all possible exceptions and treating them as warnings is the right approach here, or pretty much ever. There are specific errors we'd expect to see in the course of normal operation (like Elasticsearch not being available or authentication being rejected), but if something truly exceptional occurs, we'll want it to be treated properly as an error.

Reply:
1. this is the maps collector, so that part is right
2. I agree this shouldn't be swallowed, but on the other hand, I hope the timing errors or connection handling can be handled more by Task Manager

Reply:
The timing errors should be avoidable, by reworking the way the task manager gets initialized. So that we completely avoid the chicken or the egg problem.